File: TechniCasino/backend/src/models/game.rs
// ── models/game.rs — Game data models ──
//
// Structs:
//   Game           → game definition (id, name, game_type, min_bet, max_bet)
//   GameSession    → active game session (id, user_id, game_id, state, created_at)
//   GameResult     → completed game result (session_id, outcome, payout)
//   Bet            → individual bet (id, session_id, amount, choice)
//
// Enums:
//   GameType       → Blackjack, Roulette, Slots, Poker
//   GameOutcome    → Win, Lose, Push

File: TechniCasino/packages/shared/src/types/game.ts
// ── Game Types ──
// Define and export:
//
// GameType        → enum: BLACKJACK, ROULETTE, SLOTS, POKER, MINES, CHICKEN_ROAD
// GameInfo        → id, name, type (GameType), description, minBet, maxBet, imageUrl, isMultiplayer
// GameSession     → id, userId, gameId, state (GameState), bet, result, createdAt
// GameState       → enum: WAITING, IN_PROGRESS, COMPLETED
// GameResult      → sessionId, outcome (WIN/LOSE/PUSH), payout, multiplier, provablyFairData
// Bet             → id, sessionId, amount, choice (game-specific data as JSON)
//
// ProvablyFairData → serverSeedHash, clientSeed, nonce, serverSeed (revealed after game)
//
// Each game will have its own specific types in its own engine.ts files
